fractal_server/app/models/job.py

```diff
@@ -87,35 +86,56 @@ class Config:
         arbitrary_types_allowed = True
 
     id: Optional[int] = Field(default=None, primary_key=True)
-    project_id: int = Field(foreign_key="project.id")
-    input_dataset_id: int = Field(foreign_key="dataset.id")
-    output_dataset_id: int = Field(foreign_key="dataset.id")
-    workflow_id: int = Field(foreign_key="workflow.id")
-    working_dir: Optional[str]
-    working_dir_user: Optional[str]
-    first_task_index: int
-    last_task_index: int
 
+    project_id: Optional[int] = Field(foreign_key="project.id")
+    project: Optional["Project"] = Relationship(  # noqa: F821
+        back_populates="job_list"
+    )
+
+    workflow_id: Optional[int] = Field(foreign_key="workflow.id")
+    workflow: Optional["Workflow"] = Relationship(  # noqa: F821
+        back_populates="job_list"
+    )
+
+    input_dataset_id: Optional[int] = Field(foreign_key="dataset.id")
     input_dataset: Dataset = Relationship(
         sa_relationship_kwargs=dict(
+            back_populates="list_jobs_input",
             lazy="selectin",
             primaryjoin="ApplyWorkflow.input_dataset_id==Dataset.id",
         )
     )
+
+    output_dataset_id: Optional[int] = Field(foreign_key="dataset.id")
     output_dataset: Dataset = Relationship(
         sa_relationship_kwargs=dict(
+            back_populates="list_jobs_output",
             lazy="selectin",
             primaryjoin="ApplyWorkflow.output_dataset_id==Dataset.id",
         )
     )
-    workflow: Workflow = Relationship()
 
-    workflow_dump: Optional[dict[str, Any]] = Field(sa_column=Column(JSON))
+    user_dump: dict[str, Any] = Field(
+        sa_column=Column(JSON, nullable=False, server_default="{}")
```

Comment:
I think we don't want a server default here

Reply:
(this also applies to the other "dump" attributes)

Reply:
> Can we split the migration in two migrations (or two blocks of the same migration.. it's the same)? The first adds the column with a server default, so that it can correctly upgrade an existing DB. The second one removes the default.

I now see that this is done, in the migration, but in a way which I do not fully understand (see comment below, related to the `alter_table` parameters).